src/error.rs

```diff
@@ -108,6 +114,12 @@ impl From<httparse::Error> for Error {
     }
 }
 
+impl From<Http2Error> for Error {
+    fn from(err: Http2Error) -> Error {
+        Error::Http2(err)
```

Comment:
Could you add a `from_and_cause!` case to the tests below, to keep the coverage happy.